leku/src/main/java/com/schibstedspain/leku/search/LekuSearchFragment.java

```diff
@@ -0,0 +1,252 @@
+package com.schibstedspain.leku.search;
+
+import android.app.Activity;
+import android.content.ActivityNotFoundException;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.speech.RecognizerIntent;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.support.v7.widget.Toolbar;
+import android.text.TextWatcher;
+import android.view.LayoutInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.inputmethod.EditorInfo;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.EditText;
+import android.widget.ImageView;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.GoogleApiAvailability;
+import com.schibstedspain.leku.LocationPicker;
+import com.schibstedspain.leku.NullView;
+import com.schibstedspain.leku.R;
+import com.schibstedspain.leku.tracker.TrackEvents;
+import java.util.List;
+
+public class LekuSearchFragment extends Fragment {
+
+  private LekuSearchCallback lekuSearchCallback;
+
+  public static LekuSearchFragment newInstance() {
+    return new LekuSearchFragment();
+  }
+
+  private static final String LAST_LOCATION_QUERY = "last_location_query";
+  private static final int MIN_CHARACTERS = 2;
+  private static final int REQUEST_PLACE_PICKER = 6655;
+  private static final int CONNECTION_FAILURE_RESOLUTION_REQUEST = 9000;
+
+  private Toolbar toolbar;
+  private EditText searchView;
+  private ImageView clearSearchButton;
+  private TextWatcher textWatcher;
+  private MenuItem searchOption;
+
+  @Nullable
+  @Override
+  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+    super.onCreateView(inflater, container, savedInstanceState);
+    return inflater.inflate(R.layout.toolbar_search, null, false);
+  }
+
+  @Override
+  public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+
+    bindViews(view);
+    setupToolbar();
+    setUpClearListener();
```

Comment:
Let's rename `setupToolbar` to `setUpToolbar`, to match the common style. Or rename the others `setUp` methods to `setup`.